p2p/client.go

```diff
@@ -439,3 +581,19 @@ func (c *Client) findConnection(peer peer.AddrInfo) bool {
 	}
 	return false
 }
+
+// validates the content identifiers advertised in the DHT are valid
+type blockIdValidator struct{}
+
+func (blockIdValidator) Validate(_ string, id []byte) error {
+	var err error
+	defer func() {
+		cid.MustParse(string(id))
+		// recover from panic if one occurred. Set err to nil otherwise.
+		if recover() != nil {
+			err = errors.New("invalid cid")
+		}
+	}()
+	return err
```

Comment:
pretty sure this wont return an error ever, need named arg

Reply:
thanks, fixed it. I added also a test for it.